tests/Annotations/AnnotationsTest.php

```diff
@@ -0,0 +1,142 @@
+        $response = $client->getResponse();
+        $this->assertResponseIsValid($response);
+        $this->assertSame(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     * @dataProvider invalidPageProvider
+     */
+    public function it_returns_a_404_for_an_invalid_page(string $page)
+    {
+        $client = static::createClient();
+
+        $this->mockProfileCall($this->createProfile('1234'));
+        $this->mockHypothesisSearchCall('1234', new EmptyIterator(), 0, (int) $page);
+
+        $client->request('GET', "/annotations?by=1234&page=$page");
+        $response = $client->getResponse();
+        $this->assertResponseIsValid($response);
+
+        $this->assertSame(404, $response->getStatusCode());
+        $this->assertSame('application/problem+json', $response->headers->get('Content-Type'));
+        $this->assertResponseIsValid($response);
+        $this->assertJsonStringEqualsJson(['title' => "No page $page", 'type' => 'about:blank'], $response->getContent());
+        $this->assertFalse($response->isCacheable());
+    }
+
+    public function invalidPageProvider() : Traversable
+    {
+        foreach (['-1', '0', '2', 'foo'] as $page) {
+            yield 'page '.$page => [$page];
+        }
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_return_restricted_annotations()
+    {
+        $client = static::createClient();
+
+        $this->mockHypothesisTokenCall('1234', '1234access');
+        $this->mockHypothesisSearchCall('1234', $this->createAnnotations(), 20, 1, 20, '', 'desc', 'updated', ['Authorization' => 'Bearer 1234access']);
+        $client->request('GET', '/annotations?by=1234&access=restricted', [], [], ['HTTP_X_CONSUMER_GROUPS' => 'user,view-restricted-annotations']);
+        $response = $client->getResponse();
+        $this->assertResponseIsValid($response);
+
+        $this->assertSame(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_return_public_only_annotations()
+    {
+        $client = static::createClient();
+
+        $this->mockHypothesisSearchCall('4321', $this->createAnnotations(), 20);
+        $client->request('GET', '/annotations?by=4321&access=public', [], [], ['HTTP_X_CONSUMER_GROUPS' => 'user,view-restricted-annotations']);
+        $response = $client->getResponse();
+        $this->assertResponseIsValid($response);
+
+        $this->assertSame(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_restrict_access_to_private_annotations()
+    {
+        $client = static::createClient();
+
+        $this->mockHypothesisSearchCall('4321', $this->createAnnotations(), 20);
+        $client->request('GET', '/annotations?by=4321&access=restricted');
+        $response = $client->getResponse();
+        $this->assertResponseIsValid($response);
+
+        $this->assertSame(400, $response->getStatusCode());
+    }
```

Comment:
I think there is one missing case: `it_will_return_public_only_annotations_to_unauthenticated_users`, targeting

- `/annotations?by=4321&access=public` (or `/annotations?by=4321` if the default access is `public`)
- mock is a call to hypothesis without an authentication token
- 200 response code